File: app/src/main/java/com/githang/gradledoc/app/others/Contributor.java
package com.githang.gradledoc.app.others;

import com.alibaba.fastjson.annotation.JSONField;

/**
 * 贡献者
 *
 * @author 黄浩杭 (huanghaohang@parkingwang.com)
 * @version 2015-12-02
 * @since 2015-12-02
 */
public class Contributor {
    private String name;
    private String avatar;
    private int contributions;

    public String getName() {
        return name;
    }

    @JSONField(name = "login")
    public void setName(String name) {
        this.name = name;
    }

    public String getAvatar() {
        return avatar;
    }

    @JSONField(name = "avatar_url")
    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public int getContributions() {
        return contributions;
    }

    @JSONField(name = "contributions")
    public void setContributions(int contributions) {
        this.contributions = contributions;
    }
}

File: app/src/main/java/com/githang/gradledoc/app/others/Contributor.kt
package com.githang.gradledoc.app.others

import android.databinding.BindingAdapter
import android.widget.ImageView
import com.githang.gradledoc.common.ImageLoader
import com.squareup.moshi.Json

/**
 * 贡献者
 *
 * @author 黄浩杭 (huanghaohang@parkingwang.com)
 * @version 2015-12-02
 * @since 2015-12-02
 */
class Contributor(
        @Json(name = "login") val name: String? = null,
        @Json(name = "avatar_url") val avatar: String? = null,
        @Json(name = "contributions") val contributions: Int = 0
) {
    companion object {
        @BindingAdapter(value = ["imageUrl"])
        @JvmStatic
        fun loadImage(imageView: ImageView, url: String) {
            ImageLoader.loadImage(url, imageView)
        }
    }
}
